daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/VariableMap1.scala

```diff
@@ -407,32 +446,54 @@ class VariableMap private(vTable: Map[GlobalQName, ArrayBuffer[VariableInstance]
         }
 
         case _ => {
-          variable.setValue(VariableUtils.convert(newValue.getAnyRef.toString, variable.rd, referringContext))
-          variable.setState(VariableSet)
+          vrd.direction match {
+            /**
+             * Due to potential race conditions regarding the setting of
+             * variables via setVariable and default values in cominbation with
+             * suspesions during unparsing, we only allow the use of either
+             * setVariable statements or a default value when unparsing a
+             * variable.
+             */
+            //case VariableDirection.UnparseOnly | VariableDirection.Both if (vrd.maybeDefaultValueExpr.isDefined) => {
```

Comment:
So I think we can enable this check, but we can make it check for something narrower that will not cause existing tests to fail

If a variable has no NVIs associated with it. Then it is just a global variable, and I think this SDW/SDE is not needed at all. 

If a variable does have NVI statements, let's call that a scoped variable. If any of those NVI statements have default values, or the global declaration has a default value, then I think this warning is important if the variable direction is "both", and should be an SDE if the direction is "unparse". 

This requires us to compute whether a variable is global, or scoped with any default values supplied, if direction is both or parsed.